File: CS/NavigateCommandExample/Models/MedicalAppointment.cs
using System;

namespace SchedulerDragDropExample {
    public class MedicalAppointment {
        internal MedicalAppointment(DateTime startTime, DateTime endTime, int doctorId, int paymentStateId, string location, string patientName, string note, bool firstVisit) {
            StartTime = startTime;
            EndTime = endTime;
            DoctorId = doctorId;
            PaymentStateId = paymentStateId;
            Note = note;
            Location = location;
            PatientName = patientName;
            FirstVisit = firstVisit;
        }

        public int Id { get; set; }
        public bool AllDay { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string PatientName { get; set; }
        public string Note { get; set; }
        public string Subject { get; set; }
        public int PaymentStateId { get; set; }
        public int IssueId { get; set; }
        public int Type { get; set; }
        public string Location { get; set; }
        public string RecurrenceInfo { get; set; }
        public string ReminderInfo { get; set; }
        public int? DoctorId { get; set; }
        public bool FirstVisit { get; set; }
    }
}

File: VB/NavigateCommandExample/Models/MedicalAppointment.vb
Namespace SchedulerDragDropExample

    Public Class MedicalAppointment

        Friend Sub New(ByVal startTime As Date, ByVal endTime As Date, ByVal doctorId As Integer, ByVal paymentStateId As Integer, ByVal location As String, ByVal patientName As String, ByVal note As String, ByVal firstVisit As Boolean)
            Me.StartTime = startTime
            Me.EndTime = endTime
            Me.DoctorId = doctorId
            Me.PaymentStateId = paymentStateId
            Me.Note = note
            Me.Location = location
            Me.PatientName = patientName
            Me.FirstVisit = firstVisit
        End Sub

        Public Property Id As Integer

        Public Property AllDay As Boolean

        Public Property StartTime As Date

        Public Property EndTime As Date

        Public Property PatientName As String

        Public Property Note As String

        Public Property Subject As String

        Public Property PaymentStateId As Integer

        Public Property IssueId As Integer

        Public Property Type As Integer

        Public Property Location As String

        Public Property RecurrenceInfo As String

        Public Property ReminderInfo As String

        Public Property DoctorId As Integer?

        Public Property FirstVisit As Boolean
    End Class
End Namespace
